opus/application/static_media/js/browse.js

```diff
@@ -2210,30 +2326,83 @@ var o_browse = {
+        if (slug === opus.prefs.cols.last()) {
+            width -= $(`.op-data-table-view .ps__thumb-y`).width();
+        }
+        let top = $(elem).offset().top +
+                  $(elem).outerHeight();
+
+        o_browse.checkForEmptyMetadataList();
+
+        $("#op-edit-field-tool").css({
+            display: "block",
+            top: top,
+            left: left,
+            width: width,
+        }).slideDown("slow");
+
+        $("#op-edit-field-tool").data("slug", slug);
+    },
+
+    hideTableMetadataTools: function() {
+        $("#op-edit-field-tool").removeClass("show").hide();
+    },
+
+    onDoneUpdateFromTableMetadataDetails: function(e) {
+        o_utils.disableUserInteraction();
+
+        o_hash.updateURLFromCurrentHash(); // This makes the changes visible to the user
+        // passing in false indicates to not close the gallery view on loadData
+        if (opus.prefs.cols.length <= 1) {
+            $("a.op-metadata-detail-remove").addClass("op-button-disabled");
+        } else {
+            $("a.op-metadata-detail-remove").removeClass("op-button-disabled");
+        }
+        o_selectMetadata.saveChanges();
+        o_selectMetadata.reRender();
+        o_browse.hideMetadataList();
+        o_browse.hideTableMetadataTools();
+
+        o_utils.enableUserInteraction();
+    },
+
     hideMetadataList: function() {
         $("#op-add-metadata-fields").removeClass("show").hide();
     },
 
     adjustTopOfMetadataList: function(elem) {
-        let top = $(elem).position().top;
-        let galleryViewContentsHeight = $("#galleryViewContents").height();
-        let menuHeight = $(`#op-add-metadata-fields .op-select-list`).height();
-
-        // if the top of the dropdrown is more than half way down the list, dropup instead
-        if (top * 2 > galleryViewContentsHeight) {
-            // make sure to move the bottom to the top of the line, not the bottom
-            top -= (menuHeight + $(elem).height());
+        let top = $(elem).offset().top;
+        // if this is coming from the slideshow view, calulate the top differently
+        if ($(elem).hasClass("op-metadata-details-tools")) {
+            let galleryViewContentsHeight = $("#galleryViewContents").height();
+            let menuHeight = $(`#op-add-metadata-fields .op-select-list`).height();
+
+            // if the top of the dropdrown is more than half way down the list, dropup instead
+            if (top * 2 > galleryViewContentsHeight) {
+                // make sure to move the bottom to the top of the line, not the bottom
+                top -= (menuHeight + $(elem).height());
+            }
         }
         return top;
     },
 
     showMetadataList: function(e) {
-        let contextMenu = "#op-add-metadata-fields";
-        let targetPosition = $(e.currentTarget).position();
-        let left = targetPosition.left;
+        let targetPosition = $(e.currentTarget).offset();
```

Comment:
Something in here seems to have changed. With a little work I can get the *slideshow* add-metadata dropdown to dropup when it shouldn't, and run a little off the top of the browser. If I have exactly the same screen on the production OPUS, it doesn't do it.